provider/sts.go

```diff
@@ -13,9 +13,9 @@ func stsResource() tfsdk.NestedAttributes {
 			Optional:    true,
 			Computed:    true,
 		},
-		"oidc_private_key_secret_arn": {
-			Description: "OIDC Private Key Secret ARN",
-			Type:        types.StringType,
+		"oidc_config": {
```

Comment:
As I wrote before - I think it shouldn't be an attribute but a type.... 
```suggestion
"oidc_config_id": {
		Description: "OIDC config ID",
		Type:        types.StringType,
		Optional: true, 
```